### PR TITLE
feat: Invite command

### PR DESCRIPTION
**Pull request by theovgl**

### CODE REVIEW

**Review by ppiques — APPROVED** [review]
Impressive. Very nice. Let's see @vfurmane's review.

File: include/Channel.hpp [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 # include <stdint.h>
 # include <string>
+#include <vector>
```
Comment (vfurmane) [inline include/Channel.hpp:6]: Missing space

File: include/Channel.hpp [hunk 2]
```diff
@@ -56,6 +57,8 @@ class Channel : public _base_channel
 		void	unsetKey(void);
 		bool	compareKey(const std::string &key) const;
 		void	setCreator(User &user);
+		void	addInvitation(std::string nickname);
+		bool	isInvited(std::string nickname);
```
Comment (vfurmane) [inline include/Channel.hpp:61]: Use const reference

File: src/Channel.cpp [hunk 3]
```diff
@@ -129,6 +129,16 @@ void	Channel::setCreator(User &user)
 	user.setStatus(CHANNEL_CREATOR);
 }
 
+void	Channel::addInvitation(std::string nickName)
```
Comment (vfurmane) [inline src/Channel.cpp:132]: nick**NNNNNNNNNNNNNNNNNNNNNNNN**ame ?????

File: src/IRCReplies.cpp [hunk 4]
```diff
@@ -43,3 +43,14 @@ RPL_MYINFO::RPL_MYINFO(Peer &peer, bool include_prefix) : Message(peer, std::str
 		this->prefix = this->updatePrefixFromPeer();
 	this->input = this->updateInputFromFields();
 }
+
+RPL_INVITING::RPL_INVITING(Peer &peer, const std::string &channel, const std::string &nickname, bool include_prefix) : Message(peer, std::string())
+{
+	this->command = "341";
+	this->arguments[0] = peer.getUsername();
```
Comment (vfurmane) [inline src/IRCReplies.cpp:50]: This argument is too much

File: src/IRCReplies.cpp [hunk 5]
```diff
@@ -43,3 +43,14 @@ RPL_MYINFO::RPL_MYINFO(Peer &peer, bool include_prefix) : Message(peer, std::str
...
+	this->arguments[1] = channel + " " + nickname;
```
Comment (vfurmane) [inline src/IRCReplies.cpp:51]: This should be the 0th argument

File: src/IRCReplies.cpp [hunk 6]
```diff
@@ -43,3 +43,14 @@ RPL_MYINFO::RPL_MYINFO(Peer &peer, bool include_prefix) : Message(peer, std::str
...
+	this->argCount = 2;
```
Comment (vfurmane) [inline src/IRCReplies.cpp:52]: Just one `argCount`

File: src/commands/invite.cpp [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+	if (!_base_channel::isValidName(message.arguments[1]))
+		return 1;
+	_base_channel base_channel = _base_channel::parse(message.arguments[1]);
+	if (deps.channels.has(base_channel))
+	{
+		if (!deps.peers.hasByNickname(message.arguments[0]))
+			throw ERR_NOSUCHNICK(message.arguments[0]);
+
+		Channel target_channel(deps.channels.get(base_channel));
```
Comment (vfurmane) [inline src/commands/invite.cpp:27]:
```suggestion
		Channel &target_channel = deps.channels.get(base_channel);
```

File: src/commands/invite.cpp [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+		if (!target_channel.users.hasByNickname(message.peer.getNickname()))
+			throw ERR_NOTONCHANNEL(message.arguments[1]);
+		else
```
Comment (vfurmane) [inline src/commands/invite.cpp:31]: `else` is not necessary as you throw on the upper line

File: src/commands/invite.cpp [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+		{
+			if ((target_channel.getFlags() & FLAG_INVITE) == FLAG_INVITE)
+			{
+				if (target_channel.users.getByNickname(message.peer.getNickname()).getStatus() > 0)
```
Comment (vfurmane) [inline src/commands/invite.cpp:35]:
```suggestion
				if (target_channel.users.getByNickname(message.peer.getNickname()).getStatus() > CHANNEL_USER)
```

**Review by vfurmane — CHANGES_REQUESTED** [review]
¯\_(ツ)_/¯

File: include/Channel.hpp [hunk 10]
```diff
@@ -57,8 +57,8 @@ class Channel : public _base_channel
 		void	unsetKey(void);
 		bool	compareKey(const std::string &key) const;
 		void	setCreator(User &user);
-		void	addInvitation(std::string nickname);
-		bool	isInvited(std::string nickname);
+		void	addInvitation(const std::string nickname);
+		bool	isInvited(const std::string nickname) const;
```
Comment (vfurmane) [inline include/Channel.hpp:61]: Constant but not reference :slightly_smiling_face:

**Review by vfurmane — CHANGES_REQUESTED** [review]
You're a bastard, but not rigorous one

**Review by vfurmane — APPROVED** [review]

**Pull request merged**